src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/GraphAuthDAO.scala

```diff
@@ -78,22 +83,28 @@ class GraphAuthDAO extends AuthDAO with GraphDAO {
           .loop(
             "vtx", //start point of loop
             invert(isVertexOfClass(VertexSchema.Workspace)), //loop condition: stop walking the path once we've found a workspace node
-            isConnectedToTargetWorkspace(workspaceId)) //loop body: only emit paths to access maps connected to the target workspace
+            isVertexOfClass(VertexSchema.Workspace)) //loop body: emit paths to all workspaces
           .enablePath.path()
 
-      //The last two elements of each path are the group and then the workspace access map.
-      //The edge label between them corresponds to the access level for the group.
-      //This foldl is equivalent to doing a max() on the list of those access levels.
-      accessMapPaths.iterator.foldLeft(WorkspaceAccessLevels.NoAccess:WorkspaceAccessLevel)({ (maxAccessLevel, path) =>
-        val reversePath = path.reverseIterator
-        val accessMap = reversePath.next().asInstanceOf[Vertex]
-        val group = reversePath.next().asInstanceOf[Vertex]
-
-        val accessLabel = accessMap.getEdges(Direction.OUT).filter( _.getVertex(Direction.IN) == group ).head.getLabel
-        val accessLevel = WorkspaceAccessLevels.withName(EdgeSchema.stripEdgeRelation(accessLabel))
-
-        WorkspaceAccessLevels.max(accessLevel, maxAccessLevel)
-      })
+      //The last 3 elements of each path are the group then the workspace access map then the workspace.
+      //The edge label between the group and the workspace access map corresponds to the access level for the group.
+      //It is possible for there to be more than 1 access level for a user for a workspace
+      //Group the paths by workspace then take the max access level for each workspace
+      val accessByWorkspace = accessMapPaths.toList.groupBy(path => path.last.asInstanceOf[Vertex].getProperty[String]("workspaceId"))
```

Comment:
getProperty breaks encapsulation of the serialization code (maybe one day we'll base64encode our property names???) but i'm not sure i can recommend `loadObject[Workspace](vertex).workspaceId` as an alternative and keep my dignity